config/zfs-build.m4

```diff
@@ -152,6 +155,28 @@ AC_DEFUN([ZFS_AC_DEBUG_KMEM_TRACKING], [
 	AC_MSG_RESULT([$enable_debug_kmem_tracking])
 ])
 
+dnl #
+dnl # Disabled by default, enables INVARIANTS features in the FreeBSD kernel
+dnl # module.  This feature must be used when building for a FreeBSD kernel
+dnl # with "options INVARIANTS" in the KERNCONF and must not be used when the
+dnl # INVARIANTS option is absent.
+dnl #
```

Comment:
Is this something we can dynamically check for by default on FreeBSD so it always does the right thing?  Unless it's explicitly overridden by the config option of course.

Reply:
Yes, I can grep for it in a sysctl that gives the running kernel config. I just need to think a bit on the m4.

Reply:
The `ZFS_AC_CONFIG_USER_*` m4 macros provides some simple examples of the check behavior you could pretty easily adapt.